lib/Twig/Extensions/TokenParser/Trans.php

```diff
@@ -24,16 +24,25 @@ public function parse(Twig_Token $token)
         $count = null;
         $plural = null;
         $notes = null;
+        $body = null;
+        $with = null;
 
         if (!$stream->test(Twig_Token::BLOCK_END_TYPE)) {
-            $body = $this->parser->getExpressionParser()->parseExpression();
-        } else {
+            if ($stream->nextIf('with')) {
+                $with = $this->parser->getExpressionParser()->parseHashExpression();
+            } else {
+                $body = $this->parser->getExpressionParser()->parseExpression();
+            }
+        }
+
+        if (null === $body) {
             $stream->expect(Twig_Token::BLOCK_END_TYPE);
             $body = $this->parser->subparse(array($this, 'decideForFork'));
             $next = $stream->next()->getValue();
 
             if ('plural' === $next) {
-                $count = $this->parser->getExpressionParser()->parseExpression();
+                $countExpr = $this->parser->getExpressionParser()->parseMultitargetExpression();
+                $count = new Twig_Node_Expression_Function('abs', $countExpr, $lineno);
```

Comment:
I'm added the abs function directly as function expression to count node.
In the previous code, it was compiled as string by raw method.